clients/roscpp/src/libros/xmlrpc_manager.cpp

```diff
@@ -369,7 +372,17 @@ void XMLRPCManager::releaseXMLRPCClient(XmlRpcClient *c)
   {
     if (c == i->client_)
     {
-      i->in_use_ = false;
+      if (!shutting_down_)
```

Comment:
pedantic: I'd do `if (shutting_down_)` and swap the code for the `if` branch with the `else` one. IMO it'd read better.